synapse/handlers/federation.py

```diff
@@ -1903,12 +1900,10 @@ async def _handle_new_event(
             await self.persist_events_and_notify(
                 [(event, context)], backfilled=backfilled
             )
-            success = True
-        finally:
-            if not success:
-                run_in_background(
-                    self.store.remove_push_actions_from_staging, event.event_id
-                )
+        except Exception:
+            run_in_background(
+                self.store.remove_push_actions_from_staging, event.event_id
+            )
```

Comment:
apart from anything else, this swallows the exception... you need to re-raise.

Reply:
I think otherwise this should be ok. Worst case we'll get some confusing stacktraces and have to put it back.

Reply:
Right, thanks!